opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -155,36 +285,47 @@ def flush(self) -> None:
 
 
 class LogEmitter:
-    # TODO: Add multi_log_processor
     def __init__(
         self,
         resource: Resource,
+        multi_log_processor: Union[
+            SynchronousMultiLogProcessor, ConcurrentMultiLogProcessor
+        ],
         instrumentation_info: InstrumentationInfo,
     ):
         self._resource = resource
+        self._multi_log_processor = multi_log_processor
         self._instrumentation_info = instrumentation_info
 
     @property
     def resource(self):
         return self._resource
 
     def emit(self, record: LogRecord):
-        # TODO: multi_log_processor.emit
-        pass
+        """Emits the :class:`LogData` by associating :class:`LogRecord`
+        and instrumentation info.
+        """
+        log_data = LogData(record, self._instrumentation_info)
+        self._multi_log_processor.emit(log_data)
 
     def flush(self):
```

Comment:
Why does the `LogEmitter` have a flush method? Shouldn't the `Provider` be responsible for the processors?

Reply:
Both provider and emitter can flush. This exists to complete the `Appender/Handler` like Interfaces/ABC. In our case, `OTLPHandler` has to implement two methods (emit and flush) after subclassing `logging.Handler`. Handler will have a access to Emitter (analogous to Tracer in instrumentation). So if `.flush` of the handler is called (by high-level call `logging.shutdown()` or any other manner) we make use of `LogEmitter` flush method to flush the logs.  Does this explanation make it clear why there is a flush on the Emitter?

Spec https://github.com/open-telemetry/oteps/blob/main/text/logs/0150-logging-library-sdk.md#logemitter.

Reply:
The use case makes sense. However, I'm wondering what would happen if multiple `LogEmitter`s are created via `Provider` but all sharing the same `multilogprocessor`. If `flush` is called for a specific `LogEmitter`, would we flush the entire queue in the `multilogprocessor`, which might contain log records from different sources -> different `LogEmitter`s correct? I don't think we have this problem in tracing because only the `TracerProvider` is allowed to flush. Thoughts? Unless I am missing something.

Reply:
Right, how could one avoid it and only flush the logs from one lib, except when we create a new pipeline for each source? (that doesn't sound like a good idea). On the other hand provider flush does the same thing just flush everything out. Do you see any potential problems? This is probably worth sharing with logs SIG and see what they have to say about it?

Reply:
Yeah we should probably bring it up. I'll be out this week so if you could bring it up during the SIG and relay what is discussed that would be great.

Reply:
For now, I don't think this should block the PR, maybe just put a TODO next to `flush` ?